hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -2956,14 +2956,27 @@ func machineTemplateBuilders(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.
 			return nil
 		}
 	case hyperv1.AzurePlatform:
+		var err error
 		template = &capiazure.AzureMachineTemplate{}
+		machineTemplateSpec, err = azureMachineTemplateSpec(nodePool)
+		if err != nil {
+			return nil, nil, "", err
+		}
 		mutateTemplate = func(object client.Object) error {
 			o, _ := object.(*capiazure.AzureMachineTemplate)
-			spec, err := azureMachineTemplateSpec(hcluster, nodePool, o.Spec)
-			if err != nil {
-				return err
+			o.Spec = *machineTemplateSpec.(*capiazure.AzureMachineTemplateSpec)
+
+			// The azure api requires passing a public key. This key is randomly generated, the private portion is thrown away and the public key
+			// gets written to the template.
+			sshKey := o.Spec.Template.Spec.SSHPublicKey
```

Comment:
`o.Spec.Template.Spec.SSHPublicKey` is already overwritten at this point, so you wouldn't get the existing value. You might wanna do that before:
```go
o.Spec = *machineTemplateSpec.(*capiazure.AzureMachineTemplateSpec)
```

Reply:
I tried this and it fails to even set a value for sshKey. I reverted this PR back to what I previously had.

Reply:
what is the error you get?

Reply:
It doesn't seem to set the key at all - https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_hypershift/4606/pull-ci-openshift-hypershift-main-e2e-aks/1828828725832060928